wandb/sdk/artifacts/artifact.py

```diff
@@ -1597,7 +1596,7 @@ def get(self, name: str) -> Optional[data_types.WBValue]:
         # `artifact.download`. In the future, we should refactor the deserialization
         # pattern such that this special case is not needed.
         if wb_class == wandb.Table:
-            self.download(recursive=True)
+            self.download()
```

Comment:
I think the original reason for recursive here was this case:
* Artifact A contains a Table T which has a column of Images
* Artifact B contains a JoinedTable J which joins Table T with another Table of labels, say Table L.
* User logs artifact B. Note: the manifest of B only contains a reference to the table T file, not the Images.
* When doing `art_b.get("J")`, we download the referenced table T file, but not the images. The images are then downloaded serially instead.

At the time, we had mad the assumption that if you are downloading a Table, then it is quite likely that any referenced files (at the time, almost always another table) would depend on the entire artifact. 

This assumption may not hold anymore, or perhaps there is a really bad degenerate case here?

Reply:
Repro:
![Untitled](https://github.com/wandb/wandb/assets/127154459/771e8090-5562-481a-bd6d-71c43b014562)

Reply:
Here's a stack trace of what happens when you get the joined table:
- we build the joined table from JSON. First, we get the first table ([code](https://github.com/wandb/wandb/blob/b86aaff3851f9ab9e137f10969878a7e21827f65/wandb/data_types.py#L1239)).
- because the first table is a reference to another artifact, we fetch that artifact and get it from there ([code](https://github.com/wandb/wandb/blob/b86aaff3851f9ab9e137f10969878a7e21827f65/wandb/sdk/artifacts/artifact.py#L1592))
- we download the whole artifact ([code](https://github.com/wandb/wandb/blob/b86aaff3851f9ab9e137f10969878a7e21827f65/wandb/sdk/artifacts/artifact.py#L1600)). Again, no recursive download is needed.
![Untitled](https://github.com/wandb/wandb/assets/127154459/7efe0534-7a78-4d98-a6ef-fdcb07febd5c)

Reply:
I'm going to resolve this thread and merge the PR. If you can think of a case where recursive downloads are needed, please create a test for it.